modules/xfem/include/efa/EFAFuncs.h

```diff
@@ -0,0 +1,83 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef EFAFUNCS_H
+#define EFAFUNCS_H
+
+#include <map>
+#include <set>
+#include <vector>
+#include <algorithm>
+
+template <typename T>
+bool deleteFromMap(std::map<unsigned int, T*> &theMap, T* elemToDelete)
+{
+  bool didIt = false;
+  typename std::map<unsigned int, T*>::iterator i=theMap.find(elemToDelete->id());
+  if (i != theMap.end())
+  {
+    delete i->second;
+    theMap.erase(i);
```

Comment:
I'm hesitant to use smart pointers for this code because this algorithm has no dependencies on anything other than the standard library, and I'd like to keep it that way unless there's a really compelling reason.